src/database/query-table.c

```diff
@@ -123,6 +123,10 @@ bool init_memory_database(void)
 		}
 	}
 
+	// Attach disk database
+	if(!attach_database(memdb, NULL, config.files.database.v.s, "disk"))
+		return false;
```

Comment:
Should we through an error here?
___

The commit message says that we detach the database when FTL terminates - does it also detach when it crashes? Or is this an exception we can't handle?

Reply:
This makes `init_memory_database()` returning `false` which, in turn, causes the routine calling it (`FTL_fork_and_bind_sockets()`)  log this even:
``` c
	// Initialize in-memory databases
	if(!init_memory_database())
		log_crit("Cannot initialize in-memory database.");
```
The error will furthermore be logged independently via the SQLite3 logging callback we have installed.

Reply:
We do try to detach on a crash:

https://github.com/pi-hole/FTL/blob/710e0bb3b7ded854ae06d5bbaeb257cf667f77e6/src/signals.c#L259

https://github.com/pi-hole/FTL/blob/710e0bb3b7ded854ae06d5bbaeb257cf667f77e6/src/daemon.c#L346

https://github.com/pi-hole/FTL/blob/710e0bb3b7ded854ae06d5bbaeb257cf667f77e6/src/database/query-table.c#L141-L143